shtdlib.sh

```diff
@@ -178,6 +179,56 @@ function umask_decorator {
     return 1
 }
 
+# Bash behaviour option decorator
+# Allows changing/setting bash options for a function code block restoring
+# the original once execution of the function and it's calls is complete.
+# Requires an option name (see shopt) and a truthyness vale "true"/"false" or
+# other command that returns 0/1.
+# variables shopt_decorator_option_name and shopt_decorator_option_value
+# To use this add a line like the following (without #) as the first line of a function
+# shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return
+function shopt_decorator {
+    if [ -n "${shopt_decorator_option_value:-}" ]  && [ -n "$(shopt -o "${shopt_decorator_option_name:-}")" ] ; then
+        if [ "${FUNCNAME[0]}" != "${FUNCNAME[2]}" ] ; then
+            if shopt -qo "${shopt_decorator_option_name}" ; then
+                # Option is set
+                if ! "${shopt_decorator_option_value}" ; then
+                    # Option should not be set
+                    debug 10 "Temporarily unsetting bash option ${shopt_decorator_option_name}"
+                    shopt -uo "${shopt_decorator_option_name}"
+                else
+                    debug 10 "No need to unset ${shopt_decorator_option_name}, it's already ${shopt_decorator_option_value}"
+                fi
+                #shellcheck disable=2068
+                ${@}
+                # Set the option again in case it was unset
+                debug 10 "(Re)Setting ${shopt_decorator_option_name}"
+                shopt -so "${shopt_decorator_option_name}"
+                return 0
+            else
+                # Option is not set
+                if "${shopt_decorator_option_value}" ; then
+                    # Option should be set
+                    debug 10 "Temporarily setting bash option ${shopt_decorator_option_name}"
+                    shopt -so "${shopt_decorator_option_name}"
+                else
+                    debug 10 "No need to set ${shopt_decorator_option_name}, it's already ${shopt_decorator_option_value}"
```

Comment:
`s/set/unset/`

Reply:
nope, it's supposed to be set, not unset (shopt -so sets a value, shopt -uo unsets)